#! /bin/sh /usr/share/dpatch/dpatch-run
## 19_manpage_fixup.dpatch by Roland Rosenfeld <roland@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Convert Latin-1 char á to groff eqivalent in man page.
## DP: Quote minus signs to differenciate them from hyphens.

@DPATCH@
diff -urNad privoxy~/GNUmakefile.in privoxy/GNUmakefile.in
--- privoxy~/GNUmakefile.in
+++ privoxy/GNUmakefile.in
@@ -710,6 +710,10 @@
 	mkdir -p doc/source/temp && cd doc/source/temp && $(RM) * ;\
 	nsgmls ../privoxy-man-page.sgml  | sgmlspl ../../../utils/docbook2man/docbook2man-spec.pl &&\
 	perl -pi.bak -e 's/ <URL:.*>//; s/\[ /\[/g' privoxy.8 ;\
+	perl -pi.bak -e "s/\[ /\[/g;s/á/\\\\['a]/g" privoxy.8; \
+	perl -pi.bak -e 's/([ {])-([a-z])/$$1\\-$$2/g' privoxy.8; \
+	perl -pi.bak -e 's/ --([a-z])/ \\-\\-$$1/g' privoxy.8; \
+	perl -pi.bak -e 's/\\fB--/\\fB\\-\\-/g' privoxy.8; \
 	$(DB) ../privoxy-man-page.sgml && $(MV) -f privoxy.8 ../../../privoxy.8
 
 # For those with man2html ala RH7s.
@@ -722,6 +726,7 @@
 # Twice because my version of man2html is pulling in commas and periods in URLs.
 	$(PERL) -pi.bak -e 's/(<A.*),(">)/$$1$$2/g' tmp.html
 	$(PERL) -pi.bak -e 's,\.">,">,g' tmp.html
+	$(PERL) -pi.bak -e "s/\['a\]/\&aacute;/g" tmp.html
 # Get rid of spurious  from conversion. (How to do this with perl?)
 	$(SED) -e 's///g' tmp.html > doc/webserver/man-page/privoxy-man-page.html && $(RM) tmp.*
 else
